[sig-node] TestInvalidPodFiltered flakes with "Expected no update in channel" error

**Which test(s) are flaking**:

TestInvalidPodFiltered

**Reason for failure**:

```
--- FAIL: TestInvalidPodFiltered (0.00s)
    config_test.go:126: Expected no update in channel, Got types.PodUpdate{Pods:[]*v1.Pod{(*v1.Pod)(0xc0000b4000)}, Op:1, Source:"test"}
FAIL
```

This is flaking rarely enough that it is not caught by our CI jobs, which currently tolerate up to 2 unit test failures per run (!).

With that toleration removed in https://github.com/kubernetes/kubernetes/pull/93605, this flake has been seen (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93605/pull-kubernetes-bazel-test/1293289839906525184).

Reproducible with the following steps:

1. Build a test binary for the affected package:

```
go test -race -c ./pkg/kubelet/config
```

2. Stress the affected test using the [stress](https://godoc.org/golang.org/x/tools/cmd/stress) tool:

```
stress ./config.test -test.run TestInvalidPodFiltered
```

5. Once failures are seen, look at the logs to see the details about the error:

```sh
ls -lat $TMPDIR/go-stress*
cat $TMPDIR/go-stress...<filename>
```

/sig node
cc @kubernetes/sig-node-test-failures

--- a/pkg/kubelet/config/config_test.go
+++ b/pkg/kubelet/config/config_test.go
@@ -189,8 +189,8 @@ func TestInvalidPodFiltered(t *testing.T) {
 	channel <- podUpdate
 	expectPodUpdate(t, ch, CreatePodUpdate(kubetypes.ADD, TestSource, CreateValidPod("foo", "new")))
 
-	// add an invalid update
-	podUpdate = CreatePodUpdate(kubetypes.UPDATE, TestSource, &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo"}})
+	// add an invalid update, pod with the same name
+	podUpdate = CreatePodUpdate(kubetypes.ADD, TestSource, CreateValidPod("foo", "new"))
 	channel <- podUpdate
 	expectNoPodUpdate(t, ch)
 }

